src/Documenter.jl

```diff
@@ -512,8 +512,14 @@ function git_push(
                     gitrm_copy(target_dir, latest_dir)
                     Writers.HTMLWriter.generate_siteinfo_file(latest_dir, "latest")
                 else
-                    gitrm_copy(target_dir, stable_dir)
-                    Writers.HTMLWriter.generate_siteinfo_file(stable_dir, "stable")
+                    # only push to stable if this is the latest stable release
+                    versions = filter!(x -> occursin(Base.VERSION_REGEX, x), readdir(dirname))
+                    maxver = mapreduce(x -> VersionNumber(x), max, v"0.0.0", versions)
+                    vtag = VersionNumber(tag)
```

Comment:
I think here we do run into the problem that `tag` might not be a well-formed version number.

Reply:
Putting an `if` around this checking the regex should be good enough.

Reply:
And  not deploy at all if the tag is bad?

Reply:
I would still deploy `tagged_dir`, but not to stable. The release-x.y actually has this check already.

Reply:
I think we should just skip deployment then, since it would not be picked up in `generate_version_file` anyway if it is not a valid VersionNumber

Reply:
Fair enough, they should be consistent indeed.